update/src/workflows.ts

```diff
@@ -0,0 +1,43 @@
+// @@@SNIPSTART typescript-update-workflow
+import * as wf from '@temporalio/workflow';
+
+// Define an update that takes a number as input and returns a number.
+export const fetchAndAdd = wf.defineUpdate<number, [number]>('fetchAndAdd');
+// Define a second update that we will use to complete the workflow.
+export const done = wf.defineUpdate('done');
+
+export async function counter(): Promise<number> {
+  let count = 0;
+  let shouldComplete = false;
+
+  // Define an update validator that rejects negative inputs to the update.
+  const validator = (arg: number) => {
+    if (arg < 0) {
+      throw new Error('Argument must not be negative');
+    }
+  };
+
+  // Define the update handler implenting the fetchAndAdd operation. Note that
+  // an update can mutate workflow state, and return a value.
+  const handler = (arg: number) => {
```

Comment:
Worth also should that update handlers can be async and can sleep or run activities.

Reply:
I thought about mentioning that in the comment but decided not to, because I had heard you say that in your opinion async update handlers should be considered an advanced use case only and need to be used with care. So it seemed best to keep this introductory sample simple.

Reply:
Maybe it's worth doing another sample with an async update function when we have update-with-start for the async compensation pattern.